### DISCUSSION TITLE
Суммирование дубликатов

### DISCUSSION THREAD

**Discussion by juramebel** [totumonline/totum-ru-issues-and-discussions#140] [Как это сделать?] [answered]
Добрый день!
Для импорта спецификаций использую принцип [описанный на yutube по импорту csv](https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/32).
Поскольку в спецификации не должно быть дубликатов сделал проверку на дубли и вывод предупреждения через errorException.
Но хочется привести к такому виду:
оператор заполняет спецификацию. на этом этапе он часто меняет позиции комплектующих на аналоги из наличия, к примеру. Поэтому могут появляться дубли. 
Если есть дубли, то появляется возможность суммы кол-в в дублях объединить, например через кнопку. 
Как это сделать технически красиво?
Искать меньший id, суммировать все кол-ва в него, а потом удалять оставшиеся через delete?

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Если бы я был бухгалтер то я сделал бы так:

Нет никакого смысла экономить на `id` и можно все будет удалить и все заново заинсертить (`insert`) в эту же таблицу.

1. Те мы берем список оснований (он может быть не уникальным);
2. Делаем его уникальным через `uniq`;
3. Проходим по нему `listReplace` заполняя суммированием на основе текущей таблицы;
4. Все удаляем;
5. Вставляем наш новый с суммированием.

Но это я ленивый что пипец, поэтому я пока топик как отвеченный ставить не буду, может еще что придумается.

**Reply by juramebel** [reply]
Если будет время, закиньте, пожалуйста пример кода. 
Что то заблудился в трех соснах)

**Reply by ttm-alexeypolunin** [reply]
На самом деле не `2+2` задача:

```
a1=: while(preaction: $replace; action: $delete)

	delete: deleteList(table: $#ntn)

a2= : insertList(table: $#ntn; field: 'base' = $replace[[base]]; field: 'quantity' = $replace[[quantity]])

	~replace: listReplace(list: $unicRowList; action: "quantity" = $val; value: "value")
			val: listSum(list: $list_for_sum)
				list_for_sum: selectList(table: $#ntn; field: 'quantity'; where: 'base' = $#value[base])

		unicRowList: rowListCreate(field: "base" = $uniq)
			uniq: listUniq(list: $list)
				list: selectList(table: $#ntn; field: 'base')
```

**Reply by juramebel** [reply]
Супер.
Все получилось. 
Спасибо!
